### PR TITLE
[AD-101] Workaround hanging on close

### PR DESCRIPTION
**Pull request by gromakovsky**

### CODE REVIEW

File: lib/src/Pos/Launcher/Resource.hs [hunk 1]
```diff
@@ -444,7 +444,7 @@ createTransportTCP addrInfo = do
         Left e -> do
             logError $ sformat ("Error creating TCP transport: " % shown) e
             throwM e
-        Right transport -> return (concrete transport, liftIO $ NT.closeTransport transport)
+        Right transport -> return (concrete transport, pass {-# liftIO $ NT.closeTransport transport #-})
```
Comment (dniku) [inline lib/src/Pos/Launcher/Resource.hs]: kommentiki

Comment (int-index) [reply lib/src/Pos/Launcher/Resource.hs]: Why is this `{-#` instead of simply `{-`? It's not a pragma...

**Review by int-index — CHANGES_REQUESTED** [review]
Why is this OK not to `closeTransport`?

**Review by goodsoft — COMMENTED** [review]
Both Vty and Qt exit immediately with these changes for me.
Qt outputs the following line though:
`QObject::~QObject: Timers cannot be stopped from another thread`

File: lib/src/Pos/Launcher/Resource.hs (hunk 1 above)
Comment (gromakovsky) [reply lib/src/Pos/Launcher/Resource.hs]: Oh, I forgot the syntax :unamused: 
I should also leave issue ID and some explanation here I suppose.

**Review by gromakovsky — COMMENTED** [review]
> Why is this OK not to `closeTransport`?

I don't know what exactly `closeTransport` does, but as you can see it's used at most once: when the application stops. And nobody guarantees that it will be called anyway (just like any other clean-up action). Moreover, in case of Daedalus it's really never called, because Daedalus sends SIGTERM which is not caught, so clean-up code is not executed. And nobody died because of that so far.
Of course it's not absolutely OK not to `closeTransport`, but I find it not-so-bad for a workaround.

**Review by int-index — APPROVED** [review]

**Pull request merged**